akka-distributed-data/src/main/scala/akka/cluster/ddata/ReplicatedData.scala

```diff
@@ -37,6 +37,17 @@ trait ReplicatedData {
 
 }
 
+trait DeltaReplicatedData extends ReplicatedData {
+  type D <: ReplicatedData
```

Comment:
Question: Do we need to support delta with different type than the full state type? Example of such data type?
It makes it more complicated and it's not possible to use the delta as the full state when it is received at a node that has not seen that entry before.